--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -158,7 +158,7 @@
     private static final Logger LOGGER = LogManager.getLogger();
     public static final File USER_CACHE_FILE = new File("usercache.json");
     public static final WorldSettings DEMO_WORLD_SETTINGS = new WorldSettings("Demo World", GameType.SURVIVAL, false, Difficulty.NORMAL, false, new GameRules(), DatapackCodec.field_234880_a_);
-    protected final SaveFormat.LevelSave anvilConverterForAnvilFile;
+    public final SaveFormat.LevelSave anvilConverterForAnvilFile;
     protected final PlayerData playerDataManager;
     private final Snooper snooper = new Snooper("server", this, Util.milliTime());
     private final List<Runnable> tickables = Lists.newArrayList();
@@ -649,7 +649,7 @@
         {
             this.snooper.stop();
         }
-
+        net.kardexo.kardexotools.KardExo.stop();
         this.resourceManager.close();
 
         try
@@ -1081,7 +1081,7 @@
 
     public String getServerModName()
     {
-        return "vanilla";
+        return "KardExo";
     }
 
     public CrashReport addServerInfoToCrashReport(CrashReport report)
