--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -184,7 +_,7 @@
     public static final int f_177883_ = 29999984;
     public static final LevelSettings f_129743_ = new LevelSettings("Demo World", GameType.SURVIVAL, false, Difficulty.NORMAL, false, new GameRules(), DataPackConfig.f_45842_);
     private static final long f_177871_ = 50L;
-    protected final LevelStorageSource.LevelStorageAccess f_129744_;
+    public final LevelStorageSource.LevelStorageAccess f_129744_;
     protected final PlayerDataStorage f_129745_;
     private final Snooper f_129751_ = new Snooper("server", this, Util.m_137550_());
     private final List<Runnable> f_129752_ = Lists.newArrayList();
@@ -690,6 +_,7 @@
             this.f_129751_.m_19231_();
         }
 
+        net.kardexo.kardexotools.KardExo.stop();
         this.f_129740_.close();
 
         try
@@ -1123,7 +_,7 @@
     @DontObfuscate
     public String m_130001_()
     {
-        return "vanilla";
+        return "KardExo";
     }
 
     public SystemReport m_177935_(SystemReport p_177936_)
