--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -658,6 +658,8 @@
         {
             this.snooper.stop();
         }
+        
+        exopandora.kardexo.kardexotools.KardExo.stop();
     }
 
     public String getServerHostname()
@@ -1204,7 +1206,7 @@
 
     public String getServerModName()
     {
-        return "vanilla";
+        return "KardExo";
     }
 
     public CrashReport addServerInfoToCrashReport(CrashReport report)
