--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -161,7 +161,7 @@
     private static final Logger LOGGER = LogManager.getLogger();
     public static final File USERID_CACHE_FILE = new File("usercache.json");
     public static final WorldSettings DEMO_SETTINGS = new WorldSettings("Demo World", GameType.SURVIVAL, false, Difficulty.NORMAL, false, new GameRules(), DatapackCodec.DEFAULT);
-    protected final SaveFormat.LevelSave storageSource;
+    public final SaveFormat.LevelSave storageSource;
     protected final PlayerData playerDataStorage;
     private final Snooper snooper = new Snooper("server", this, Util.getMillis());
     private final List<Runnable> tickables = Lists.newArrayList();
@@ -653,6 +653,7 @@
             this.snooper.interrupt();
         }
 
+        net.kardexo.kardexotools.KardExo.stop();
         this.resources.close();
 
         try
@@ -1084,7 +1085,7 @@
 
     public String getServerModName()
     {
-        return "vanilla";
+        return "KardExo";
     }
 
     public CrashReport fillReport(CrashReport p_71230_1_)
