diff -r -U 3 minecraft_server\net\minecraft\command\ServerCommandManager.java minecraft_server_patched\net\minecraft\command\ServerCommandManager.java
--- minecraft_server\net\minecraft\command\ServerCommandManager.java
+++ minecraft_server_patched\net\minecraft\command\ServerCommandManager.java
@@ -1,5 +1,6 @@
 package net.minecraft.command;
 
+import exopandora.kardexo.kardexotools.KardExo;
 import net.minecraft.command.server.CommandBanIp;
 import net.minecraft.command.server.CommandBanPlayer;
 import net.minecraft.command.server.CommandBroadcast;
@@ -107,6 +121,8 @@
             this.registerCommand(new CommandListPlayers());
             this.registerCommand(new CommandWhitelist());
             this.registerCommand(new CommandSetPlayerTimeout());
+            
+            KardExo.getCommands().forEach(this::registerCommand);
         }
         else
         {
