--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -198,6 +_,7 @@
             GameProfileCache.m_11004_(this.m_129797_());
             f_139598_.info("Preparing level \"{}\"", (Object)this.m_7123_());
             this.m_130006_();
+            net.kardexo.kardexotools.KardExo.init(this);
             long j = Util.m_137569_() - i;
             String s = String.format(Locale.ROOT, "%.3fs", (double)j / 1.0E9D);
             f_139598_.info("Done ({})! For help, type \"help\"", (Object)s);
