--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -189,6 +189,7 @@
             PlayerProfileCache.setUsesAuthentication(this.usesAuthentication());
             LOGGER.info("Preparing level \"{}\"", (Object)this.getLevelIdName());
             this.loadLevel();
+            net.kardexo.kardexotools.KardExo.init(this);
             long j = Util.getNanos() - i;
             String s = String.format(Locale.ROOT, "%.3fs", (double)j / 1.0E9D);
             LOGGER.info("Done ({})! For help, type \"help\"", (Object)s);
