--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -189,6 +189,7 @@
             PlayerProfileCache.setOnlineMode(this.isServerInOnlineMode());
             LOGGER.info("Preparing level \"{}\"", (Object)this.func_230542_k__());
             this.func_240800_l__();
+            net.kardexo.kardexotools.KardExo.init(this);
             long j = Util.nanoTime() - i;
             String s = String.format(Locale.ROOT, "%.3fs", (double)j / 1.0E9D);
             LOGGER.info("Done ({})! For help, type \"help\"", (Object)s);
