--- a/net/minecraft/world/biome/BiomeContainer.java
+++ b/net/minecraft/world/biome/BiomeContainer.java
@@ -17,7 +17,7 @@
     public static final int BIOMES_SIZE = 1 << WIDTH_BITS + WIDTH_BITS + HEIGHT_BITS;
     public static final int HORIZONTAL_MASK = (1 << WIDTH_BITS) - 1;
     public static final int VERTICAL_MASK = (1 << HEIGHT_BITS) - 1;
-    private final Biome[] biomes;
+    public final Biome[] biomes;
 
     public BiomeContainer(Biome[] biomesIn)
     {
