--- a/net/minecraft/world/level/chunk/ChunkBiomeContainer.java
+++ b/net/minecraft/world/level/chunk/ChunkBiomeContainer.java
@@ -21,7 +_,7 @@
     private static final int f_62106_ = (1 << f_62109_) - 1;
     public static final int f_156118_ = 1 << f_62109_ + f_62109_ + DimensionType.f_156649_ - 2;
     private final IdMap<Biome> f_62111_;
-    private final Biome[] f_62112_;
+    public final Biome[] f_62112_;
     private final int f_156119_;
     private final int f_156120_;
 
