--- a/net/minecraft/server/management/PlayerInteractionManager.java
+++ b/net/minecraft/server/management/PlayerInteractionManager.java
@@ -246,6 +246,11 @@
 
     public boolean tryHarvestBlock(BlockPos pos)
     {
+    	return exopandora.kardexo.kardexotools.veinminer.Veinminer.mine(pos, this.player, this.world, this::tryHarvestBlock);
+    }
+
+    public boolean tryHarvestBlock(BlockPos pos, boolean dropAtPlayer)
+    {
         BlockState blockstate = this.world.getBlockState(pos);
 
         if (!this.player.getHeldItemMainhand().getItem().canPlayerBreakBlockWhileHolding(blockstate, this.world, pos, this.player))
