--- a/net/minecraft/server/management/PlayerInteractionManager.java
+++ b/net/minecraft/server/management/PlayerInteractionManager.java
@@ -259,6 +259,11 @@
 
     public boolean tryHarvestBlock(BlockPos pos)
     {
+    	return exopandora.kardexo.kardexotools.veinminer.Veinminer.mine(pos, this.player, this.world, this::tryHarvestBlock);
+    }
+    
+    public boolean tryHarvestBlock(BlockPos pos, boolean dropAtPlayer)
+    {
         BlockState blockstate = this.world.getBlockState(pos);
 
         if (!this.player.getHeldItemMainhand().getItem().canPlayerBreakBlockWhileHolding(blockstate, this.world, pos, this.player))
@@ -302,7 +307,7 @@
                     if (flag && flag1)
                     {
                         ItemStack itemstack1 = itemstack.isEmpty() ? ItemStack.EMPTY : itemstack.copy();
-                        block.harvestBlock(this.world, this.player, pos, blockstate, tileentity, itemstack1);
+                        block.harvestBlock(this.world, this.player, dropAtPlayer ? this.player.getPosition() : pos, blockstate, tileentity, itemstack1);
                     }
 
                     return true;
