--- a/net/minecraft/server/management/PlayerInteractionManager.java
+++ b/net/minecraft/server/management/PlayerInteractionManager.java
@@ -274,7 +274,7 @@
         }
     }
 
-    public boolean tryHarvestBlock(BlockPos pos)
+    public boolean tryHarvestBlock(BlockPos pos) {return net.kardexo.kardexotools.property.PropertyHelper.canHarvestBlock(this.player, pos) && net.kardexo.kardexotools.veinminer.Veinminer.mine(pos, this.player, this.world, this::tryHarvestBlock);} public boolean tryHarvestBlock(BlockPos pos, boolean dropAtPlayer)
     {
         BlockState blockstate = this.world.getBlockState(pos);
 
@@ -319,7 +319,7 @@
 
                     if (flag && flag1)
                     {
-                        block.harvestBlock(this.world, this.player, pos, blockstate, tileentity, itemstack1);
+                        block.harvestBlock(this.world, this.player, dropAtPlayer ? this.player.func_233580_cy_() : pos, blockstate, tileentity, itemstack1);
                     }
 
                     return true;
@@ -409,6 +409,7 @@
 
             if (!flag1)
             {
+            	if(!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithBlock(playerIn, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ActionResultType actionresulttype = blockstate.onBlockActivated(worldIn, playerIn, handIn, blockRaytraceResultIn);
 
                 if (actionresulttype.isSuccessOrConsume())
@@ -420,6 +421,7 @@
 
             if (!stackIn.isEmpty() && !playerIn.getCooldownTracker().hasCooldown(stackIn.getItem()))
             {
+            	if(!net.kardexo.kardexotools.property.PropertyHelper.canPlaceBlock(playerIn, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ItemUseContext itemusecontext = new ItemUseContext(playerIn, handIn, blockRaytraceResultIn);
                 ActionResultType actionresulttype1;
 
