--- a/net/minecraft/server/management/PlayerInteractionManager.java
+++ b/net/minecraft/server/management/PlayerInteractionManager.java
@@ -274,7 +274,7 @@
         }
     }
 
-    public boolean tryHarvestBlock(BlockPos pos)
+    public boolean tryHarvestBlock(BlockPos pos) {return net.kardexo.kardexotools.property.PropertyHelper.canHarvestBlock(this.player, pos) && net.kardexo.kardexotools.veinminer.Veinminer.mine(pos, this.player, this.world, this::tryHarvestBlock);} public boolean tryHarvestBlock(BlockPos pos, boolean dropAtPlayer)
     {
         BlockState blockstate = this.world.getBlockState(pos);
 
@@ -315,7 +315,7 @@
                     ItemStack itemstack = this.player.getHeldItemMainhand();
                     ItemStack itemstack1 = itemstack.copy();
                     boolean flag1 = this.player.func_234569_d_(blockstate);
-                    itemstack.onBlockDestroyed(this.world, blockstate, pos, this.player);
+                    itemstack.onBlockDestroyed(this.world, blockstate, dropAtPlayer ? this.player.getPosition() : pos, this.player);
 
                     if (flag && flag1)
                     {
@@ -409,6 +409,7 @@
 
             if (!flag1)
             {
+                if(!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithBlock(playerIn, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ActionResultType actionresulttype = blockstate.onBlockActivated(worldIn, playerIn, handIn, blockRaytraceResultIn);
 
                 if (actionresulttype.isSuccessOrConsume())
@@ -420,6 +421,7 @@
 
             if (!stackIn.isEmpty() && !playerIn.getCooldownTracker().hasCooldown(stackIn.getItem()))
             {
+                if(!net.kardexo.kardexotools.property.PropertyHelper.canPlaceBlock(playerIn, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ItemUseContext itemusecontext = new ItemUseContext(playerIn, handIn, blockRaytraceResultIn);
                 ActionResultType actionresulttype1;
 
