--- a/net/minecraft/server/management/PlayerInteractionManager.java
+++ b/net/minecraft/server/management/PlayerInteractionManager.java
@@ -274,7 +274,7 @@
         }
     }
 
-    public boolean destroyBlock(BlockPos p_180237_1_)
+    public boolean destroyBlock(BlockPos p_180237_1_) {return net.kardexo.kardexotools.property.PropertyHelper.canHarvestBlock(this.player, p_180237_1_) && net.kardexo.kardexotools.veinminer.Veinminer.mine(p_180237_1_, this.player, this.level, this::destroyBlock);} public boolean destroyBlock(BlockPos p_180237_1_, boolean dropAtPlayer)
     {
         BlockState blockstate = this.level.getBlockState(p_180237_1_);
 
@@ -319,7 +319,7 @@
 
                     if (flag && flag1)
                     {
-                        block.playerDestroy(this.level, this.player, p_180237_1_, blockstate, tileentity, itemstack1);
+                        block.playerDestroy(this.level, this.player, dropAtPlayer ? this.player.blockPosition() : p_180237_1_, blockstate, tileentity, itemstack1);
                     }
 
                     return true;
@@ -409,6 +409,7 @@
 
             if (!flag1)
             {
+                if(!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithBlock(p_219441_1_, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ActionResultType actionresulttype = blockstate.use(p_219441_2_, p_219441_1_, p_219441_4_, p_219441_5_);
 
                 if (actionresulttype.consumesAction())
@@ -420,6 +421,7 @@
 
             if (!p_219441_3_.isEmpty() && !p_219441_1_.getCooldowns().isOnCooldown(p_219441_3_.getItem()))
             {
+                if(!net.kardexo.kardexotools.property.PropertyHelper.canPlaceBlock(p_219441_1_, blockpos, blockstate)) {return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.player);}
                 ItemUseContext itemusecontext = new ItemUseContext(p_219441_1_, p_219441_4_, p_219441_5_);
                 ActionResultType actionresulttype1;
 
