--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1532,6 +_,7 @@
                 {
                     private void m_143678_(InteractionHand p_143679_, ServerGamePacketListenerImpl.EntityInteraction p_143680_)
                     {
+                        if (!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithEntity(ServerGamePacketListenerImpl.this.f_9743_, entity)) return;
                         ItemStack itemstack = ServerGamePacketListenerImpl.this.f_9743_.m_21120_(p_143679_).m_41777_();
                         InteractionResult interactionresult = p_143680_.m_143694_(ServerGamePacketListenerImpl.this.f_9743_, entity, p_143679_);
 
