--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1537,6 +_,7 @@
                     }
                     public void m_141994_()
                     {
+                        if (!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithEntity(ServerGamePacketListenerImpl.this.f_9743_, entity)) return;
                         if (!(entity instanceof ItemEntity) && !(entity instanceof ExperienceOrb) && !(entity instanceof AbstractArrow) && entity != ServerGamePacketListenerImpl.this.f_9743_)
                         {
                             ServerGamePacketListenerImpl.this.f_9743_.m_5706_(entity);
