--- a/net/minecraft/network/play/ServerPlayNetHandler.java
+++ b/net/minecraft/network/play/ServerPlayNetHandler.java
@@ -1493,7 +1493,7 @@
         {
             double d0 = 36.0D;
 
-            if (this.player.distanceToSqr(entity) < 36.0D)
+            if (this.player.distanceToSqr(entity) < 36.0D && net.kardexo.kardexotools.property.PropertyHelper.canInteractWithEntity(this.player, entity, p_147340_1_.getAction()))
             {
                 Hand hand = p_147340_1_.getHand();
                 ItemStack itemstack = hand != null ? this.player.getItemInHand(hand).copy() : ItemStack.EMPTY;
