--- a/net/minecraft/network/play/ServerPlayNetHandler.java
+++ b/net/minecraft/network/play/ServerPlayNetHandler.java
@@ -1390,7 +1390,7 @@
         {
             double d0 = 36.0D;
 
-            if (this.player.getDistanceSq(entity) < 36.0D)
+            if (this.player.getDistanceSq(entity) < 36.0D && net.kardexo.kardexotools.property.PropertyHelper.canInteractWithEntity(this.player, entity, packetIn.getAction()))
             {
                 Hand hand = packetIn.getHand();
                 Optional<ActionResultType> optional = Optional.empty();
