--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -272,7 +_,7 @@
         }
     }
 
-    public boolean m_9280_(BlockPos p_9281_)
+    public boolean m_9280_(BlockPos p_9281_) {return net.kardexo.kardexotools.property.PropertyHelper.canHarvestBlock(this.f_9245_, p_9281_) && net.kardexo.kardexotools.veinminer.Veinminer.mine(p_9281_, this.f_9245_, this.f_9244_, this::destroyBlock);} public boolean destroyBlock(BlockPos p_9281_, boolean dropAtPlayer)
     {
         BlockState blockstate = this.f_9244_.m_8055_(p_9281_);
 
@@ -317,7 +_,7 @@
 
                     if (flag && flag1)
                     {
-                        block.m_6240_(this.f_9244_, this.f_9245_, p_9281_, blockstate, blockentity, itemstack1);
+                        block.m_6240_(this.f_9244_, this.f_9245_, dropAtPlayer ? this.f_9245_.m_142538_() : p_9281_, blockstate, blockentity, itemstack1);
                     }
 
                     return true;
@@ -407,6 +_,7 @@
 
             if (!flag1)
             {
+                if (!net.kardexo.kardexotools.property.PropertyHelper.canInteractWithBlock(p_9266_, blockpos, blockstate)) return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.f_9245_);
                 InteractionResult interactionresult = blockstate.m_60664_(p_9267_, p_9266_, p_9269_, p_9270_);
 
                 if (interactionresult.m_19077_())
@@ -418,6 +_,7 @@
 
             if (!p_9268_.m_41619_() && !p_9266_.m_36335_().m_41519_(p_9268_.m_41720_()))
             {
+                if (!net.kardexo.kardexotools.property.PropertyHelper.canPlaceBlock(p_9266_, blockpos, blockstate)) return net.kardexo.kardexotools.property.PropertyHelper.cancelBlockInteraction(this.f_9245_);
                 UseOnContext useoncontext = new UseOnContext(p_9266_, p_9269_, p_9270_);
                 InteractionResult interactionresult1;
 
